plugins/discovery-ec2/qa/amazon-ec2/build.gradle

```diff
@@ -64,61 +66,66 @@ yamlRestTest.enabled = false
+    classpath = yamlRestTestSourceSet.getRuntimeClasspath()
   }
-  SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
-  SourceSet yamlRestTestSourceSet = sourceSets.getByName(YamlRestTestPlugin.SOURCE_SET_NAME)
-  "yamlRestTest${action}" {
-    setTestClassesDirs(yamlRestTestSourceSet.getOutput().getClassesDirs())
-    setClasspath(yamlRestTestSourceSet.getRuntimeClasspath())
+
+  tasks.named("check").configure {
+    dependsOn(yamlRestTestTask)
   }
-  check.dependsOn("yamlRestTest${action}")
 
-  testClusters."yamlRestTest${action}" {
+  testClusters.matching { it.name == yamlRestTestTask.name}.configureEach {
     numberOfNodes = ec2NumberOfNodes
     plugin ':plugins:discovery-ec2'
 
     setting 'discovery.seed_providers', 'ec2'
     setting 'network.host', '_ec2_'
-    setting 'discovery.ec2.endpoint', { "http://${-> fixture.addressAndPort}" }, IGNORE_VALUE
+    setting 'discovery.ec2.endpoint', { "http://${-> fixture.get().addressAndPort}" }, IGNORE_VALUE
 
-    systemProperty "com.amazonaws.sdk.ec2MetadataServiceEndpointOverride", { "http://${-> fixture.addressAndPort}" }, IGNORE_VALUE
+    systemProperty "com.amazonaws.sdk.ec2MetadataServiceEndpointOverride", { "http://${-> fixture.get().addressAndPort}" }, IGNORE_VALUE
   }
 }
 
 // Extra config for KeyStore
-testClusters.yamlRestTestKeyStore {
+testClusters.matching { it.name == "yamlRestTestKeyStore" }.configureEach {
   keystore 'discovery.ec2.access_key', 'ec2_integration_test_access_key'
   keystore 'discovery.ec2.secret_key', 'ec2_integration_test_secret_key'
 }
 
 // Extra config for EnvVariables
-testClusters.yamlRestTestEnvVariables {
+testClusters.matching { it.name == "yamlRestTestEnvVariables" }.configureEach {
   environment 'AWS_ACCESS_KEY_ID', 'ec2_integration_test_access_key'
   environment 'AWS_SECRET_ACCESS_KEY', 'ec2_integration_test_secret_key'
 }
 
 // Extra config for SystemProperties
-testClusters.yamlRestTestSystemProperties {
+testClusters.matching { it.name == "yamlRestTestSystemProperties" }.configureEach {
   systemProperty 'aws.accessKeyId', 'ec2_integration_test_access_key'
   systemProperty 'aws.secretKey', 'ec2_integration_test_secret_key'
 }
 
 // Extra config for ContainerCredentials
-ec2FixtureContainerCredentials.env 'ACTIVATE_CONTAINER_CREDENTIALS', true
+tasks.named("ec2FixtureContainerCredentials").configure {
+  env 'ACTIVATE_CONTAINER_CREDENTIALS', true
+}
 
-testClusters.yamlRestTestContainerCredentials {
+testClusters.matching { it.name == "yamlRestTestContainerCredentials" }.configureEach {
   environment 'AWS_CONTAINER_CREDENTIALS_FULL_URI',
     { "http://${-> tasks.findByName("ec2FixtureContainerCredentials").addressAndPort}/ecs_credentials_endpoint" }, IGNORE_VALUE
 }
 
 // Extra config for InstanceProfile
-ec2FixtureInstanceProfile.env 'ACTIVATE_INSTANCE_PROFILE', true
+tasks.named("ec2FixtureInstanceProfile").configure {
+  env 'ACTIVATE_INSTANCE_PROFILE', true
+}
```

Comment:
Does your editor not insert a trailing newline?

Reply:
nope. should it?